Facebook comments plugin & git deployment docs update

RichardJECooke opened the pull request:
- Added Facebook comments plugin to list of plugins
  - Added section to git deployment troubleshooting to explain how to fix uppercase/lowercase git merge errors in tags
  - Add line to git deployment process to actually do the deploy (forgot that step in my last commit :-\

RichardJECooke commented:
Helloooo, anyone gonna approve this?

tomap commented:
LGTM, appart from the doc where you go from 2. To 4. (No 3.).
Also, you reference hexobcli in node_modules while most people have it installed in global, thus hexo (without path) will work

RichardJECooke commented:
Thanks. I corrected the numbering and also added the option for global Hexo. (Creating a project installs Hexo locally afaik so both methods will work)

I really don't like sites advising installing things in NPM globally, it's a messy antiquated practice. As soon as you have to work on two projects with different versions of a package, or collaborate with someone else who has different packages installed globally to you, things start to break. Not to mention that you can longer just clone a repo and say `npm update`, it's extra work to install global dependencies too. Having all packages neatly inside a repo avoids dependency hell that package managers were invented to overcome in the first place. Ok, enough soapboxing.
